compiler/src/iree/compiler/Dialect/LinalgExt/Transforms/AggregatedOpInterfaceImpl.cpp

```diff
@@ -325,6 +372,11 @@ OnlineAttentionOp::decomposeOperation(OpBuilder &b) {
                                                offset);
   }
 
+  // S += mask
+  if (mask.has_value() && mask.value() != nullptr) {
+    s = applyMask(b, loc, sMap, *getMaskMap(), s, *mask);
+  }
```

Comment:
This is weird. If we have to check has_value and value != nullptr, we have something redundant. We should have only one check.

Reply:
This is because the value can be a null_value since it is optional value as dictated by the generated optional TD.

Reply:
Another way is to try get rid of the optional wrapper and just check if the value is null_value

Reply:
You can do `if (type* maskVal = mask.value_or(nullptr))` and then use `maskVal in `applyMask`

Reply:
Sounds good